### PR TITLE
Add useTranslation hook

### PR DESCRIPTION
**Pull request by AdityaKhatri**

### CODE REVIEW

File: app/Base/configs/lang.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
+export const commonLang = {
```
Comment (tnagorra) [inline app/Base/configs/lang.ts:1]: Any reason we have these spaces after nepali words?

**Review by tnagorra — APPROVED** [review]

File: app/Base/context/LanguageContext.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    },
+];
+
+export const langKeySelector = (d: LangOption) => d.key;
+export const langLabelSelector = (d: LangOption) => d.label;
+
+export default createContext<LanguageContextInterface>({
+    lang: 'en',
+    setLang: (lang: Lang) => {
+        console.warn('Trying to set language', lang);
```
Comment (frozenhelium) [inline app/Base/context/LanguageContext.ts]:
```suggestion
        console.error('Trying to set language before the language context was initialized', lang);
```

File: app/Base/hooks/useTranslation.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { useMemo, useContext } from 'react';
+
+import LanguageContext, { Lang } from '../context/LanguageContext';
+
+type Trans<T> = {
```
Comment (frozenhelium) [inline app/Base/hooks/useTranslation.ts]:
```suggestion
type Translations<T> = {
```

**Review by frozenhelium — CHANGES_REQUESTED** [review]
Lets only use `strings` for translated strings so that it's easier to statically analyze code base and find out the usage

Eg: `const strings = useTranslation(homePage);`

**Comment by AdityaKhatri** [comment]
> Lets only use `strings` for translated strings so that it's easier to statically analyze code base and find out the usage
> 
> Eg: `const strings = useTranslation(homePage);`

@frozenhelium Makes sense, but we might have to use 2 objects in the same page and that creates an issue.

**Review by subinasr — APPROVED** [review]

**Comment by frozenhelium** [comment]
@AdityaKhatri we can still do
```
const homeStrings = useTranslation(homePage);
const commonStrings = useTranslation(commonLang);
const strings = {
    ...homeStrings,
    ...commonStrings,
};
```

**Pull request merged**